### DISCUSSION TITLE
Generator info integreation

### DISCUSSION THREAD

**Discussion by Lilian-Muxart** [tolwi/hassio-ecoflow-cloud#383] [Show and tell]
## Smart Generator Dual Fuel

*Sensors*
- Fuel tank (`pd.oilVal`)
- Working duration of generator (`pd.motorUseTime`)
- Total power (`pd.totalPower`)
- Remaining time (`pd.remainTime`)
- DC power (`pd.dcPower`)
- DC state (`pd.dcState`)
- DC (12V) Charge Current(`pd.dcCur`)
- DC out state (`pd.dcOutState`)
- DC Voltage (`pd.dcVol`)
- Oil max power (`pd.oilMaxOutPower`)
- AC power (`pd.acPower`)
- AC state (`pd.acState`)
- AC Voltage (`pd.acVol`)
- AC Charge Current(`pd.acCur`)
- System Mode (`pd.sysMode`)
- Error Code (`pd.errCode`)
- Motor state (`pd.motorState`)
- Temperature (`pd.temp`)
- Software Version (`pd.ver`)

*Switches*
- Motor Enabled ( `pd.motorState` -> `{"moduleType": 2, "operateType": "motorCtrl", "params": {"enabled": "VALUE"}}`)
- Mode Eco ( `pd.sysMode` -> `{"moduleType": 2, "operateType": "modeCtrl", "params": {"mode": "VALUE"}}`)

*Sliders (numbers)*
- DC Discharge Speed ( `pd.oilMaxOutPower` -> `{"moduleType": 2, "operateType": "oilMaxOutPower", "params": {"oilMax": VALUE}}` [1000 - 1800])

**Comment by sauron3256** [comment]
Following with interest. Any patches to test?

**Reply by Lilian-Muxart** [reply]
> Following with interest. Any patches to test?

I have finally developed a module for the dual-fuel smart generator that works.

**Reply by sauron3256** [reply]
I'd love to provide test data if you need me to.

**Reply by Lilian-Muxart** [reply]
I am going to request to publish my code, and you will be able to test it to see if there are any adjustments to be made.

**Comment by Lilian-Muxart** [comment]
I managed to retrieve all the values and modify them via MQTT, but now the part I haven’t looked into at all is integrating what I found into Home Assistant. That’s why I’m posting this here, in case someone manages to integrate it.
